tests/integration/security/sds_tls_origination/egress_gateway_origination_test.go

```diff
@@ -0,0 +1,293 @@
+				credNameMissing = "tls-credential-not-created-cacert"
+			)
+
+			var credentialA = sdstlsutil.TLSCredential{
+				CaCert: sdstlsutil.MustReadCert(t, "root-cert.pem"),
+			}
+			var CredentialB = sdstlsutil.TLSCredential{
+				CaCert: sdstlsutil.FakeRoot,
+			}
+			// Add kubernetes secret to provision key/cert for gateway.
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{credName}, "SIMPLE", credentialA, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{credName})
+
+			// Add kubernetes secret to provision key/cert for gateway.
+			sdstlsutil.CreateKubeSecret(t, ctx, []string{fakeCredName}, "SIMPLE", CredentialB, false)
+			defer ingressutil.DeleteKubeSecret(t, ctx, []string{fakeCredName})
+
+			internalClient, externalServer, _, serverNamespace := sdstlsutil.SetupEcho(t, ctx)
+
+			// Set up Host Namespace
+			host := "server." + serverNamespace.Name() + ".svc.cluster.local"
+
+			testCases := map[string]struct {
+				response        []string
+				credentialToUse string
+				gateway         bool // true if the request is expected to be routed through gateway
+			}{
+				// Use CA certificate stored as k8s secret with the same issuing CA as server's CA.
+				// This root certificate can validate the server cert presented by the echoboot server instance.
+				"Simple TLS with Correct Root Cert": {
+					response:        []string{response.StatusCodeOK},
+					credentialToUse: strings.TrimSuffix(credName, "-cacert"),
+					gateway:         true,
+				},
+				// Use CA certificate stored as k8s secret with different issuing CA as server's CA.
+				// This root certificate cannot validate the server cert presented by the echoboot server instance.
+				"Simple TLS with Fake Root Cert": {
+					response:        []string{response.StatusCodeUnavailable},
+					credentialToUse: strings.TrimSuffix(fakeCredName, "-cacert"),
+					gateway:         false,
+				},
+
+				// Set up an UpstreamCluster with a CredentialName when secret doesn't even exist in istio-system ns.
+				// Secret fetching error at Gateway, results in a 503 response.
+				"Simple TLS with credentialName set when the underlying secret doesn't exist": {
+					response:        []string{response.StatusCodeUnavailable},
+					credentialToUse: strings.TrimSuffix(credNameMissing, "-cacert"),
+					gateway:         false,
+				},
+			}
+
+			for name, tc := range testCases {
+				t.Run(name, func(t *testing.T) {
+					bufDestinationRule := sdstlsutil.CreateDestinationRule(t, serverNamespace, "SIMPLE", tc.credentialToUse)
+
+					// Get namespace for gateway pod.
+					istioCfg := istio.DefaultConfigOrFail(t, ctx)
+					systemNS := namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+
+					time.Sleep(time.Second * 5)
+
+					ctx.Config().ApplyYAMLOrFail(ctx, systemNS.Name(), bufDestinationRule.String())
+					defer ctx.Config().DeleteYAMLOrFail(ctx, systemNS.Name(), bufDestinationRule.String())
+
+					time.Sleep(time.Second * 5)
```

Comment:
Could you add a comment for the reason of this sleep? Ideally it is better to use a check of CDS update with retry, or sending requests with longer retry timeout instead of adding the sleep here.

Reply:
I've already removed this sleep in favor of a longer timeout